src/libraries/System.Text.Json/gen/Reflection/TypeWrapper.cs

```diff
@@ -396,12 +396,6 @@ public override PropertyInfo[] GetProperties(BindingFlags bindingAttr)
             {
                 if (item is IPropertySymbol propertySymbol)
                 {
-                    // Skip auto-generated properties on records.
-                    if (_typeSymbol.IsRecord && propertySymbol.DeclaringSyntaxReferences.Length == 0)
```

Comment:
What was the original intent of skipping - not to add properties if they aren't used (within the current assembly anyway)?

Reply:
In previous iterations of property discovery, there were a bunch of implicit properties on records that were being added for source generation that are never intended for source generation. This was adding significant size to user apps. Verified that these props are no longer being included even with this change.